src/main/java/com/wb/utility/Translator.java

```diff
@@ -0,0 +1,23 @@
+package com.wb.utility;
+
+import java.util.Locale;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.i18n.LocaleContextHolder;
+import org.springframework.context.support.ResourceBundleMessageSource;
+import org.springframework.stereotype.Component;
+
+@Component
+public class Translator {
+    private static ResourceBundleMessageSource messageSource;
```

Comment:
`final` zamiast `static`

Reply:
Tutaj jest static ponieważ metoda toLocale jest metodą statyczną a wewnątrz metody użyty jest obiekt ResourceBundleMessageSource i ta metoda statyczna jest używana bez tworzenia obiektu

Reply:
jeśli metoda jest statyczna to może zostać użyta bez istniejącej instancji danego obiektu
message source ustawiasz wtedy kiedy tworysz instancje
czyli z jednej strony pozwalasz na uzycie metody bez instacji a z drugiej storny bez stworzenia instancji bedziesz mial nulla w message source. Translator powinien zostac komponentem a ta metoda powinna nie byc satyczna (tak samo jak to pole). Translatora musisz wstrzykiwac tam gdzie chcesz przez konstruktor jak kazdy inny bean